graph/iterator/and_iterator.go

```diff
@@ -255,13 +276,25 @@ func (it *And) cleanUp() {}
 // Close this iterator, and, by extension, close the subiterators.
 // Close should be idempotent, and it follows that if it's subiterators
 // follow this contract, the And follows the contract.
-func (it *And) Close() {
+//
+// Note: as this potentially involves closing multiple subiterators, only
```

Comment:
Don't add this as a note, just say "It closes all sub iterators it can, but returns the first error it encounters." (cribbed from os.RemoveAll).

Also below where this is used.